File: ConversorDeUnidadesPy/ConversorDeUnidades/conversor_unidades.py
def mostrar_menu():
    print("Bienvenidos al Conversor de Unidades")
    print("1. Metros a Kilómetros")
    print("2. Kilómetros a Metros")
    print("3. Celsius a Fahrenheit")
    print("4. Fahrenheit a Celsius")
    print("5. Kilogramos a Libras")
    print("6. Libras a Kilogramos")
    print("7. Salir")

def convertir_metros_a_kilometros(metros):
    return metros / 1000
def convertir_kilometros_a_metros(kilometros):
    return kilometros * 1000
def convertir_celsius_a_fahrenheit(celsius):
    return (celsius * 9/5) + 32
def convertir_fahrenheit_a_celsius(fahrenheit):
    return (fahrenheit - 32) * 5/9
def convertir_kilogramos_a_libras(kilogramos):
    return kilogramos * 2.20462
def convertir_libras_a_kilogramos(libras):
    return libras / 2.20462

def main():
    while True:
        mostrar_menu()
        opcion = input("Selecciona una opción (1-7): ")

        if opcion == "1":
            metros = float(input("Ingresa la cantidad en metros: "))
            print(f"{metros} metros son {convertir_metros_a_kilometros(metros)} kilómetros.")
        elif opcion == "2":
            kilometros = float(input("Ingresa la cantidad en kilómetros: "))
            print(f"{kilometros} kilómetros son {convertir_kilometros_a_metros(kilometros)} metros.")
        elif opcion == "3":
            celsius = float(input("Ingresa la temperatura en Celsius: "))
            print(f"{celsius}°C son {convertir_celsius_a_fahrenheit(celsius)}°F.")
        elif opcion == "4":
            fahrenheit = float(input("Ingresa la temperatura en Fahrenheit: "))
            print(f"{fahrenheit}°F son {convertir_fahrenheit_a_celsius(fahrenheit)}°C.")
        elif opcion == "5":
            kilogramos = float(input("Ingresa el peso en kilogramos: "))
            print(f"{kilogramos} kg son {convertir_kilogramos_a_libras(kilogramos)} lb.")
        elif opcion == "6":
            libras = float(input("Ingresa el peso en libras: "))
            print(f"{libras} lb son {convertir_libras_a_kilogramos(libras)} kg.")
        elif opcion == "7":
            print("¡Gracias por usar el Conversor de Unidades!")
            break
        else:
            print("Opción no válida. Por favor, intenta de nuevo.")
        print()

if __name__=="__main__":
    main()

File: ConversorDeUnidadesJava/ConversorUnidades/src/conversorunidades/ConversorUnidades.java

package conversorunidades;

import java.util.Scanner;

public class ConversorUnidades {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        int opcion;
        
        do {
            System.out.println("\n---Conversor de Unidades---");
            System.out.println("1. Metros a Kilómetros");
            System.out.println("2. Kilómetros a Metros");
            System.out.println("3. Celsius a Fahrenheit");
            System.out.println("4. Fahrenheit a Celsius");
            System.out.println("5. Kilogramos a Libras");
            System.out.println("6. Libras a Kilogramos");
            System.out.println("7. Salir");
            opcion = entrada.nextInt();
            
            switch (opcion) {
                case 1:
                    convertirMetrosAKilometros(entrada);
                    break;
                case 2:
                    convertirKilometrosAMetros(entrada);
                    break;
                case 3:
                    converirCelsiusAFahrenheit(entrada);
                    break;
                case 4:
                    convertirFahrenheitACelsius(entrada);
                    break;
                case 5:
                    convertirKilogramosALibras(entrada);
                    break;
                case 6:
                    convertirLibrasAKilogramos(entrada);
                    break;
                case 7:
                    System.out.println("Saliendo del programa...");
                    break;
                default:
                    System.out.println("Opción no válida. Intente nuevamente.");
            }
        } while (opcion != 7);  
        entrada.close();
    }
    
    public static void convertirMetrosAKilometros(Scanner entrada) {
        System.out.println("Ingrese la cantidad en metros: ");
        double metros = entrada.nextDouble();
        double kilometros = metros / 1000;
        System.out.printf("%.2f m son %.2f km.%n", metros, kilometros);
    }
    
    public static void convertirKilometrosAMetros(Scanner entrada) {
        System.out.println("Ingrese la cantidad en kilómetros: ");
        double kilometros = entrada.nextDouble();
        double metros = kilometros * 1000;
        System.out.printf("%.2f km son %.2f m.%n", kilometros, metros);
    }
    
    public static void converirCelsiusAFahrenheit(Scanner entrada) {
        System.out.println("Ingrese la temperatura en Celsius: ");
        double celsius = entrada.nextDouble();
        double fahrenheit = (celsius * 9/5) + 32;
        System.out.printf("%.2f °C son %.2f °F.%n", celsius, fahrenheit);
    }
    
    public static void convertirFahrenheitACelsius(Scanner entrada) {
        System.out.println("Ingrese la temperatura en Fahrenheit: ");
        double fahrenheit = entrada.nextDouble();
        double celsius = (fahrenheit - 32) * 5/9;
        System.out.printf("%.2f °F son %.2f °C.%n", fahrenheit, celsius);
    }
    
    public static void convertirKilogramosALibras(Scanner entrada) {
        System.out.println("Ingrese el peso en kilogramos: ");
        double kilogramos = entrada.nextDouble();
        double libras = kilogramos * 2.20462;
        System.out.printf("%.2f kg son %.2f lb.%n", kilogramos, libras);
    }
    
    public static void convertirLibrasAKilogramos(Scanner entrada) {
        System.out.println("Ingresa el peso en libras: ");
        double libras = entrada.nextDouble();
        double kilogramos = libras / 2.20462;
        System.out.printf("%.2f lb son %.2f kg.%n", libras, kilogramos);
    }
}
